lib/aws-record/record/batch_writer.rb

```diff
@@ -0,0 +1,208 @@
+# Copyright 2015-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+module Aws
+  module Record
+    class BatchWriter < Batch
+      include Enumerable
+
+      # Writer states
+      PENDING_STATE = 'PENDING'.freeze
```

Comment:
I don't think that it would be useful to have those since this is a blocking call. I also can't see a scenario where another thread would eve want to know what the status of batch_writer is. This is namely for inspection of whether it is in a state that can be sent or whether all items sent were successful